src/hotspot/share/runtime/globals.hpp

```diff
@@ -835,6 +835,13 @@ const int ObjectAlignmentInBytes = 8;
           "JVM aborts, producing an error log and core/mini dump, on the "  \
           "first occurrence of an out-of-memory error thrown from JVM")     \
                                                                             \
+  product(intx, UserThreadWaitAttemptsAtExit, 30,                           \
+          "Number of attempts waiting for user threads in native during "   \
+          "JVM exit. Each wait attempt is 10-millisecond. The max allowed " \
+          "wait attempts for user threads in native is 1000, which is "     \
+          "10 seconds.")                                                    \
```

Comment:
Also I just realized that it makes sense to limit the user thread attempts to the internal compiler thread maximum - otherwise we would still stop waiting once the compiler thread maximum was reached.

Reply:
> Suggestion:
> 
> ```
> The number of times to wait for user threads to stop executing native code
> during JVM exit. Each wait lasts 10 milliseconds. The maximum number of waits
> is 1000, to wait at most 10 seconds.
> ```

The suggested edit looks good to me. Incorporated, thanks.

Reply:
> Also I just realized that it makes sense to limit the user thread attempts to the internal compiler thread maximum - otherwise we would still stop waiting once the compiler thread maximum was reached.

`max_wait_attempts_compiler_thread` is 1000.